packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalogItem.tsx

```diff
@@ -0,0 +1,117 @@
+import TextFieldsIcon from '@mui/icons-material/TextFields';
+import Crop75Icon from '@mui/icons-material/Crop75';
+import ArrowDropDownCircleIcon from '@mui/icons-material/ArrowDropDownCircle';
+import LayersIcon from '@mui/icons-material/Layers';
+import DnsIcon from '@mui/icons-material/Dns';
+import ContactPageIcon from '@mui/icons-material/ContactPage';
+import TabIcon from '@mui/icons-material/Tab';
+import TuneIcon from '@mui/icons-material/Tune';
+import ToggleOnIcon from '@mui/icons-material/ToggleOn';
+import RadioButtonCheckedIcon from '@mui/icons-material/RadioButtonChecked';
+import DateRangeIcon from '@mui/icons-material/DateRange';
+import CheckBoxIcon from '@mui/icons-material/CheckBox';
+import DashboardCustomizeSharpIcon from '@mui/icons-material/DashboardCustomizeSharp';
+import AddIcon from '@mui/icons-material/Add';
+import { SvgIconProps } from '@mui/material/SvgIcon';
+
+const iconMap = new Map<string | RegExp, React.ComponentType<SvgIconProps>>([
+  ['Button', SmartButtonIcon],
+  ['Image', ImageIcon],
+  ['DataGrid', GridOnIcon],
+  ['Typography', TextFieldsIcon],
+  ['TextField', Crop75Icon],
+  ['Select', ArrowDropDownCircleIcon],
+  ['Paper', LayersIcon],
+  ['Form', DnsIcon],
+  ['Card', ContactPageIcon],
+  ['Tabs', TabIcon],
+  ['Slider', TuneIcon],
+  ['Switch', ToggleOnIcon],
+  ['Radio', RadioButtonCheckedIcon],
+  ['DatePicker', DateRangeIcon],
+  ['Checkbox', CheckBoxIcon],
+  ['CodeComponent', DashboardCustomizeSharpIcon],
+  ['CreateNew', AddIcon],
+]);
+
+type ComponentItemKinds = 'future' | 'builtIn' | 'create' | 'custom';
+
+interface ComponentIconProps {
+  id: string;
+  kind?: ComponentItemKinds;
+}
+
+const ComponentIcon = ({ id: componentId, kind }: ComponentIconProps) => {
+  const Icon = iconMap.get(kind === 'custom' ? 'CodeComponent' : componentId);
+  return Icon ? <Icon fontSize="medium" opacity={kind === 'future' ? 0.75 : 1} /> : null;
+};
+
+interface ComponentCatalogItemProps {
+  draggable?: boolean;
+  onDragStart?: (event: React.DragEvent<HTMLDivElement>) => void;
+  onClick?: () => void;
+  builtIn?: string;
+  id: string;
+  displayName: string;
+  kind?: ComponentItemKinds;
+}
+
+const ComponentCatalogItem = ({
+  draggable,
+  onClick,
+  id,
+  displayName,
+  builtIn,
+  kind,
+  onDragStart,
+}: ComponentCatalogItemProps) => {
+  return (
+    <Box
+      className="ComponentCatalogItem"
+      onClick={onClick}
+      draggable={draggable}
+      onDragStart={onDragStart}
+      sx={{
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        justifyContent: 'center',
+        rowGap: 1,
+        width: builtIn ? 75 : 70,
+        height: builtIn ? 75 : 70,
+        padding: 1,
+        borderRadius: 1,
+        border: 1,
+        borderColor: 'divider',
+        borderStyle: kind === 'create' ? 'dashed' : 'solid',
+        color: 'text.secondary',
+        '&:hover': {
+          backgroundColor: 'action.disabled',
```

Comment:
This looks right to me:

```suggestion
          backgroundColor: 'action.hover',
```

It adds a black background with opacity, so it should be just additive with whatever is underneath. For future reference, I'd say adding `action.disabled` to a hover effect can be categorized as a code smell. And if we have to do it to make things look right, that'd probably be something that needs to fix in the theme. Image a maintainer working on the theme in 6 months from now finding out we've been using `action.disabled` for all kinds of things other than disabled states.